### PR TITLE
Enable confirm

### PR DESCRIPTION
**Pull request by robob27**
Sister PR to [#2721](https://github.com/DFHack/dfhack/pull/2721) to enable confirm

- Rename `gui/confirm-opts` to `gui/confirm`
- Use a Window/ZScreen in `gui/confirm`
- Add `confirm` to System Services in control-panep

### CODE REVIEW

File: gui/confirm.lua [hunk 1]
```diff
@@ -1,7 +1,7 @@
 -- confirm plugin options
 --[====[
 
-gui/confirm-opts
+gui/confirm
 ================
 A basic configuration interface for the `confirm` plugin.
```
Comment (myk002) [inline gui/confirm.lua:7]: you can remove this old copy of the docs. it's no longer used and is just liable to get out of sync with the real docs in the rst

File: gui/confirm.lua [hunk 2]
```diff
@@ -12,14 +12,11 @@ confirm = require 'plugins.confirm'
 gui = require 'gui'
 widgets = require 'gui.widgets'
 
-Opts = defclass(Opts, gui.FramedScreen)
+Opts = defclass(Opts, widgets.Window)
 Opts.ATTRS = {
     frame_style = gui.GREY_LINE_FRAME,
```
Comment (myk002) [inline gui/confirm.lua:17]: you should use the default window style here (i.e. remove this line)

File: gui/confirm.lua [hunk 3]
```diff
@@ -31,47 +28,30 @@ function Opts:init()
...
         widgets.Label{
             view_id = 'controls',
             frame = {b = 0, l = 0},
             text = {
-                {key = 'SELECT', text = ': Toggle, '},
-                {key = 'LEAVESCREEN', text = ': Back', on_activate = self:callback('dismiss')},
-                NEWLINE,
-                {key = 'SEC_SELECT', text = ': Toggle all'},
+                {key = 'SELECT', text = ': Toggle'},
             },
```
Comment (myk002) [inline gui/confirm.lua:41]: To allow this hotkey hint to be clickable too, it should be a HotkeyLabel that runs `toggle` on SELECT

**Comment by myk002** [comment]
Ok, so next steps:
1. Convert confirm dialogs to Lua-drawn widget.Windows with standard theming and clickable widgets
2. ignore unfocused ZScreens when determining which screen should be queried for a focus string (I'll take care of this)
3. moar confirm checks (at your leisure)

did I miss anything that we talked about?

File: gui/confirm.lua [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+            text_pen = COLOR_GREY,
+            cursor_pen = COLOR_WHITE,
+            choices = {},
+            on_submit = self:callback('toggle'),
+        },
+        widgets.HotkeyLabel{
+            frame = {b=1, l=0},
+            label='Toggle all',
+            key='CUSTOM_ALT_E',
+            on_activate=function() self:toggle_all(self.subviews.list:setSelected(i)) self:refresh() end,
```
Comment (myk002) [inline gui/confirm.lua:27]: what's `i` set to here?
actually, looking at how this is used, this is supposed to be getSelected(), isn't it?

Comment (robob27) [reply gui/confirm.lua]: Yeahhh, just fixed that up

**Comment by robob27** [comment]
> Ok, so next steps:
> 
>     1. Convert confirm dialogs to Lua-drawn widget.Windows with standard theming and clickable widgets
> 
>     2. ignore unfocused ZScreens when determining which screen should be queried for a focus string (I'll take care of this)
> 
>     3. moar confirm checks (at your leisure)
> 
> 
> did I miss anything that we talked about?

I can't think of anything, other than maybe if we were making changes to what I did with the getSelectedStockpile stuff. There is a bit more cleaning up to do in Gui.cpp as well I think in general but nothing show stopping for now I think 🤞

File: gui/confirm.lua [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+    end
+    self.subviews.list:setChoices(choices)
+end
+
+function Opts:choice_pen(index, enabled)
+    return (enabled and COLOR_GREEN or COLOR_RED) + (index == self.subviews.list:getSelected() and 8 or 0)
+end
+
+function Opts:toggle(choice)
+    choice = self.data[choice]
```
Comment (myk002) [inline gui/confirm.lua:89]:
```suggestion
function Opts:toggle(idx)
    local choice = self.data[idx]
```

Comment (myk002) [reply gui/confirm.lua:89]: since `choice` is normally reserved for the object table in the list item

**Review by myk002 — APPROVED** [review]

**Comment by myk002** [comment]
bah, I forgot to list config persistence. from discord: ```
One thing that's still needed for confirm is config persistence. It can go in a separate PR, though
User config should be stored in dfhack-config/confirm.json and not be associated with the current fort since the choices are global
```

**Comment by robob27** [comment]
So many things to consider for this stuff! Oopsie

**Pull request merged**